components/Diary.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
+import { Text, TouchableOpacity } from "react-native";
+import { SafeAreaView } from "react-native-safe-area-context";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+import * as ImagePicker from "expo-image-picker";
+
+import styled from "styled-components/native";
+
+export default function Diary({ navigation, route }) {
+  const [diaryInfo, setDiaryInfo] = useState({
+    content: "",
+    imageUrl: "",
+    date: "",
+  });
+
+  const openImagePicker = async () => {
+    const permissionResult =
+      await ImagePicker.requestMediaLibraryPermissionsAsync();
+
+    if (!permissionResult.granted) {
+      alert("Please allow permission!");
+
+      return;
+    }
+
+    const pickerResult = await ImagePicker.launchImageLibraryAsync();
+
+    if (pickerResult.cancelled) return;
+
+    setDiaryInfo({
+      ...diaryInfo,
+      imageUrl: pickerResult.uri,
+      date: new Date().toLocaleDateString(),
+      id: new Date().getTime(),
+    });
+  };
+
+  const deleteImage = () => {
+    setDiaryInfo({
+      ...diaryInfo,
+      imageUrl: null,
+    });
+  };
+
+  const handleChangeText = (txt) => {
+    setDiaryInfo({
+      ...diaryInfo,
+      content: txt,
+    });
+  };
+
+  const saveDiaryData = async () => {
+    if (!diaryInfo.content || !diaryInfo.imageUrl) {
+      alert("사진과 내용 모두 적어주세요.");
+      return;
+    }
+
+    try {
+      let diaryData = JSON.parse(await AsyncStorage.getItem("Diary-data"));
+
+      if (!diaryData) {
+        diaryData = [];
+      }
```

Comment:
이렇게 바꿔봐도 좋을 것 같아요
[MDN](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)

```suggestion
      let diaryData = JSON.parse(await AsyncStorage.getItem("Diary-data")) || [];
```